testkit/src/main/scala/akka/kafka/testkit/internal/TestcontainersKafka.scala

```diff
@@ -50,13 +51,35 @@ object TestcontainersKafka {
 
     def zookeeperContainer: GenericContainer[_] = cluster.getZooKeeper
 
+    def schemaRegistryContainer: Option[SchemaRegistryContainer] = cluster.getSchemaRegistry.asScala
+
+    def getSchemaRegistryUrl: String =
+      cluster.getSchemaRegistry.asScala
+        .map(_.getSchemaRegistryUrl)
+        .getOrElse(
+          throw new RuntimeException("Did you enable schema registry in your KafkaTestkitTestcontainersSettings?")
+        )
+
+    /**
+     * Deprecated: Starts Kafka, but does not override with provided settings.
+     */
+    @deprecated("Use startKafkaWithSettings instead", "2.0.3")
     def startKafka(settings: KafkaTestkitTestcontainersSettings): String = {
```

Comment:
The next version is 2.0.4.
Why was the new `startKafkaWithSettings` needed?

Reply:
This method implied that the provided settings would be used, but it ignored the `settings` param and applied settings from the overloaded `testcontainersSettings` property.

```
def startKafka(settings: KafkaTestkitTestcontainersSettings): String = {
  val settings = testcontainersSettings
```

I'm not sure why I did this originally, I think it was just a bug that slipped under the radar.

`startWithKafkaSettings` uses the provided settings.  I deprecated this method for backwards compatibility, and added an overload `startKafka()` which uses the overloaded `testcontainersSettings`.  Alternatively, instead of deprecating I could just fix the bug.

Reply:
Just fix that bug. It will do what the method promised in the first place and we do not promise strict bin-compatibility for testkits.

Reply:
Ah, ok.